d3a_interface/matching_algorithms/pay_as_clear_matching_algorithm.py

```diff
@@ -52,19 +52,20 @@ def __init__(self):
     def get_matches_recommendations(self, matching_data: Dict) -> List:
         """Returns the recommended bid offer matches"""
         matches = []
-        for market_id, data in matching_data.items():
-            bids = data.get("bids")
-            offers = data.get("offers")
-            current_time = data.get("current_time")
-            clearing = self.get_clearing_point(bids, offers, current_time, market_id)
-            if clearing is None:
-                return []
-
-            if clearing.energy > 0:
-                log.info(f"Market Clearing Rate: {clearing.rate} "
-                         f"||| Clearing Energy: {clearing.energy} ")
-            matches.extend(self._create_bid_offer_matches(
-                self.sorted_offers, self.sorted_bids, market_id, current_time))
+        for market_id, time_slot_data in matching_data.items():
+            for time_slot, data in time_slot_data.items():
+                bids = data.get("bids")
+                offers = data.get("offers")
+                current_time = data.get("current_time")
+                clearing = self.get_clearing_point(bids, offers, current_time, market_id)
+                if clearing is None:
+                    return []
```

Comment:
```suggestion
                    return
```

I know it was there before your code changes, but just caught my attention, better to return None as `clearing ` is a dataclass of `Clearing`, so rather than return an empty list, which itself is a different data format then `Clearing`

Reply:
@Faizan this method should return a list as in the type hints, the `clearing` variable doesn't have anything to do with the return value (We're returning the `matches` list)